lib/nodes/orchestra/signal.cpp

```diff
@@ -0,0 +1,221 @@
+    return orchestra::SignalType::BOOLEAN;
+  default:
+    throw RuntimeError("Unsupported signal type: {}", signalTypeToString(t));
+  }
+}
+
+node::SignalType
+villas::node::orchestra::toNodeSignalType(orchestra::SignalType t) {
+  switch (t) {
+  case orchestra::SignalType::FLOAT32:
+  case orchestra::SignalType::FLOAT64:
+    return node::SignalType::FLOAT;
+
+  case orchestra::SignalType::INT8:
+  case orchestra::SignalType::UINT8:
+  case orchestra::SignalType::INT16:
+  case orchestra::SignalType::UINT16:
+  case orchestra::SignalType::INT32:
+  case orchestra::SignalType::UINT32:
+  case orchestra::SignalType::INT64:
+  case orchestra::SignalType::UINT64:
+    return node::SignalType::INTEGER;
+
+  case orchestra::SignalType::BOOLEAN:
+    return node::SignalType::BOOLEAN;
+
+  default:
+    throw RuntimeError("Unsupported Orchestra signal type: {}", (int)t);
+  }
+}
+
+orchestra::SignalType
+villas::node::orchestra::signalTypeFromString(const std::string &t) {
+  if (t == "boolean") {
+    return orchestra::SignalType::BOOLEAN;
+  } else if (t == "unsigned int8") {
+    return orchestra::SignalType::UINT8;
+  } else if (t == "unsigned int16") {
+    return orchestra::SignalType::UINT16;
+  } else if (t == "unsigned int32") {
+    return orchestra::SignalType::UINT32;
+  } else if (t == "unsigned int64") {
+    return orchestra::SignalType::UINT64;
+  } else if (t == "int8") {
+    return orchestra::SignalType::INT8;
+  } else if (t == "int16") {
+    return orchestra::SignalType::INT16;
+  } else if (t == "int32") {
+    return orchestra::SignalType::INT32;
+  } else if (t == "int64") {
+    return orchestra::SignalType::INT64;
+  } else if (t == "float32") {
+    return orchestra::SignalType::FLOAT32;
+  } else if (t == "float64") {
+    return orchestra::SignalType::FLOAT64;
+  } else if (t == "bus") {
+    return orchestra::SignalType::BUS;
+  } else {
+    throw RuntimeError("Unknown Orchestra signal type: {}", t);
+  }
+}
+
+std::string
+villas::node::orchestra::signalTypeToString(orchestra::SignalType t) {
+  switch (t) {
+  case orchestra::SignalType::BOOLEAN:
+    return "boolean";
+  case orchestra::SignalType::UINT8:
+    return "unsigned int8";
+  case orchestra::SignalType::UINT16:
+    return "unsigned int16";
+  case orchestra::SignalType::UINT32:
+    return "unsigned int32";
+  case orchestra::SignalType::UINT64:
+    return "unsigned int64";
+  case orchestra::SignalType::INT8:
+    return "int8";
+  case orchestra::SignalType::INT16:
+    return "int16";
+  case orchestra::SignalType::INT32:
+    return "int32";
+  case orchestra::SignalType::INT64:
+    return "int64";
+  case orchestra::SignalType::FLOAT32:
+    return "float32";
+  case orchestra::SignalType::FLOAT64:
+    return "float64";
+  case orchestra::SignalType::BUS:
+    return "bus";
+  default:
+    throw RuntimeError("Unknown Orchestra signal type: {}", (int)t);
+  }
+}
+
+SignalData
+villas::node::orchestra::toNodeSignalData(const char *orchestraData,
```

Comment:
if `orchestraData` is opaque data, shouldn't it be `const void*`? Also a good candidate for a smart pointer.

Reply:
This function is also not checking the size or if `orchestraData` is nullptr

Reply:
> if orchestraData is opaque data, shouldn't it be const void*?

Fixed